include/sys/icu.h

```diff
@@ -0,0 +1,34 @@
+#ifndef _ICU_METHODS_H_
+#define _ICU_METHODS_H_
+
+#include <sys/device.h>
+#include <sys/resource.h>
+#include <sys/interrupt.h>
+
+typedef struct icu_methods {
+  void (*intr_setup)(device_t *dev, resource_t *irq, ih_filter_t *filter,
+                     ih_service_t *service, void *arg, const char *name);
+  void (*intr_teardown)(device_t *dev, resource_t *irq);
+} icu_methods_t;
+
+#define ICU_METHODS(dev) ((icu_methods_t *)(dev)->driver->interfaces[DIF_ICU])
+
+static inline void icu_intr_setup(device_t *dev, resource_t *irq,
```

Comment:
I don't think this is a good candidate for `static inline` function. If it has more complex control flow than just single `if` statement it should be moved to `.c` file.